lib/src/authentication/services/auth_service.dart

```diff
@@ -1,13 +1,80 @@
 import "package:firebase_auth/firebase_auth.dart";
+import "package:google_sign_in/google_sign_in.dart";
+import "package:kana_to_kanji/src/locator.dart";
+import "package:logger/logger.dart";
 
 /// [AuthService] Manage all the authentications processes related to firebase.
 /// It's for 2 objectives :
 /// - Isolate the code to have a better reusability
 /// - Facilitate unit tests
 class AuthService {
+  final Logger _logger = locator<Logger>();
+
+  final GoogleSignIn _googleSignIn = GoogleSignIn();
+  final AppleAuthProvider _appleAuthProvider = AppleAuthProvider();
+
+  Future<User?> getCurrentUser() async =>
+      Future.value(FirebaseAuth.instance.currentUser);
+
   /// Sign in anonymously using Firebase.
-  /// Returns a UserCredential which contains all the information of the current
-  /// user, its token...
-  Future<UserCredential> signInAnonymously() =>
-      FirebaseAuth.instance.signInAnonymously();
+  /// returns : Future<UserCredential?> - Contains all the information of the
+  /// current user, its token...
+  /// The UserCredential is null if an error occurs.
+  Future<UserCredential?> signInAnonymously() async {
+    try {
+      return await FirebaseAuth.instance.signInAnonymously();
+    } on FirebaseAuthException catch (e) {
+      _logger.e(e);
+      return Future.value();
+    }
+  }
+
+  /// Sing in with Google.
+  /// returns : Future<UserCredential?> - Contains all the information of the
+  /// current user, its token...
+  /// The UserCredential is null if an error occurs.
+  Future<UserCredential?> signInWithGoogle() async {
+    try {
+      // Trigger the authentication flow
+      final GoogleSignInAccount? googleUser = await _googleSignIn.signIn();
+
+      // Obtain the auth details from the request
+      final GoogleSignInAuthentication? googleAuth =
+          await googleUser?.authentication;
+
+      // Create a new credential
+      final credential = GoogleAuthProvider.credential(
+        accessToken: googleAuth?.accessToken,
+        idToken: googleAuth?.idToken,
+      );
+
+      // Once signed in, return the UserCredential
+      return await FirebaseAuth.instance.signInWithCredential(credential);
+    } on FirebaseAuthException catch (e) {
+      _logger.e(e);
+      return Future.value();
+    }
+  }
+
+  /// Sing in with Apple.
+  /// returns : Future<UserCredential?> - Contains all the information of the
+  /// current user, its token...
+  /// The UserCredential is null if an error occurs.
+  Future<UserCredential> signInWithApple() async =>
```

Comment:
Did you go through [this tutorial](https://firebase.google.com/docs/auth/ios/apple#configure-sign-in-with-apple) to enable Apple? To know, if I need to create a ticket

Reply:
No I followed this documentation you gave in the ticket : https://firebase.google.com/docs/auth/flutter/federated-auth?_gl=1*1qycdvu*_up*MQ..*_ga*OTk5NzYwOTk3LjE3MjA2MTE0MDM.*_ga_CW55HF8NVT*MTcyMDYxMTQwMy4xLjAuMTcyMDYxMTQwMy4wLjAuMA..#apple

The documentation you mentionned is for Swift or Objective-C. So I won't be able to use it.

So I followed the instructions, but I haven't tested as I can't test IOS stuff

Reply:
I will try the iOS and Android build today after work, it's the last thing I have to do before approving it